api/v1alpha4/machinedeployment_types.go

```diff
@@ -53,6 +53,10 @@ const (
 	// is machinedeployment.spec.replicas + maxSurge. Used by the underlying machine sets to estimate their
 	// proportions in case the deployment has surge replicas.
 	MaxReplicasAnnotation = "machinedeployment.clusters.x-k8s.io/max-replicas"
+
+	// DefaultMachineDeploymentUniqueLabelKey is the label applied to Machines
+	// in a MachineDeployment containing the hash of the template.
+	DefaultMachineDeploymentUniqueLabelKey = "machine-template-hash"
```

Comment:
I wonder why this constant has a `Default` prefix. I would also drop the `Key` suffix.

I assume this PR is meant for the v1alpha4 and v1beta1 branches, right? (and on v1beta1 we can then drop the old package)

I just wonder if we want to deprecate the mdutil package with v0.4.x and drop it with v1.0.0 or if we want to deprecate it with v1.0.0 and drop it with v1.1.0

Reply:
>I assume this PR is meant for the v1alpha4 and v1beta1 branches, right? (and on v1beta1 we can then drop the old package)

Right that was the intention of the PR, so no need to keep this in v1beta1.